client/blocks/product-purchase-features-list/customize-theme.jsx

```diff
@@ -1,24 +1,39 @@
 import { localize } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
 import customizeImage from 'calypso/assets/images/illustrations/dashboard.svg';
+import QueryActiveTheme from 'calypso/components/data/query-active-theme';
 import PurchaseDetail from 'calypso/components/purchase-detail';
+import withBlockEditorSettings from 'calypso/data/block-editor/with-block-editor-settings';
+import getCustomizeUrl from 'calypso/state/selectors/get-customize-url';
+import { getActiveTheme } from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
-function getCustomizeLink( selectedSite ) {
-	return '/customize/' + selectedSite.slug;
-}
+const CustomizeTheme = ( { translate, blockEditorSettings, areBlockEditorSettingsLoading } ) => {
+	const isFSEActive = blockEditorSettings?.is_fse_active;
+	const siteId = useSelector( getSelectedSiteId );
+	const currentThemeId = useSelector( getActiveTheme );
+	const customizeLink = useSelector( ( state ) =>
+		getCustomizeUrl( state, currentThemeId, siteId, isFSEActive )
+	);
 
-export default localize( ( { selectedSite, translate } ) => {
 	return (
 		<div className="product-purchase-features-list__item">
+			<QueryActiveTheme siteId={ siteId } />
```

Comment:
Minor question for @Automattic/team-calypso ... I notice that `<Query...` type components are usually called from higher up the component tree.

In this case we're at `/plans/my-plan`, and this is the first component that needs data about the active theme, hence introducing `<QueryActiveTheme>` here. Is there a case for moving it higher up, or does keeping it here make more sense?

Reply:
What is the query doing for us here? Ensuring up-to-date `blockEditorSettings` are fetched or something else?

Reply:
>(is that data not already populated?)

I think I may have answered my own question 😆 , I would have expected that to already be loaded at first glance.

Reply:
@creativecoder and I tested removing/re-adding the `< QueryActiveTheme />` and decided to not remove it. This will help us follow the contract for the `getThemeCustomizeUrl` function and possibly prevent future bugs related to changes revolving around the `themeId`